File: pset6/crack.py
import cs50
import sys
import crypt

def main():
    if len(sys.argv) != 2:
        print("Usage: python crack.py hash")
        exit(1)
    test_hash=sys.argv[1]

    alpha_range= list(range(65, 91)) + list(range (97, 123))
    for i in alpha_range:
        pass_hash= crypt.crypt(chr(i), "50")
        if test_hash == pass_hash:
            print("Match, password is {}",format(chr(i)))

    for i in alpha_range:
        for j in alpha_range:
            pass_hash= crypt.crypt(chr(i)+chr(j),"50")
            if test_hash == pass_hash:
                print("Match, password is {}".format(chr(i)+chr(j)))

    for i in alpha_range:
        for j in alpha_range:
            for k in alpha_range:
                pass_hash= crypt.crypt(chr(i)+chr(j)+chr(k),"50")
                if test_hash == pass_hash:
                    print("Match, password is {}".format(chr(i)+chr(j)+chr(k)))

    for i in alpha_range:
        for j in alpha_range:
            for k in alpha_range:
                for l in alpha_range:
                    pass_hash= crypt.crypt(chr(i)+chr(j)+chr(k)+chr(l),"50")
                    if test_hash == pass_hash:
                        print("Match, password is {}".format(chr(i)+chr(j)+chr(k)+chr(l)))

if __name__ == "__main__":
    main()

File: pset2/crack.c
#define _XOPEN_SOURCE
#include <unistd.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc!=2)
    {
    printf("Usage: ./crack hash");
    return 1;
    }
    else
    {}
    for (int j=2; j<6; j++)
    {
        char password[j];
        password[j-1]='\0';
        for (int l=0;l<j; l++)
        {
            for (int i=65; i<123; i++)
            {
                password[l]=(char) i;
                string hash=crypt(password, "50");
                printf("%c",password[0]);
                printf("%s\n",hash);
                if (strcmp(hash,argv[1])==0)
                {
                    printf("Correct\n");
                    break;
                }
                else
                {
                    printf("Wrong\n");
                }
            }
        }
    }
}